server/Models/getRestaurantInfo.js

```diff
@@ -1,66 +1,74 @@
 const pg = require('../db');
+const redis = require('../db/redisConnection');
 
-const getRestaurantInfo = (req, res) => {
+const getRestaurantInfo = async (req, res) => {
   const { restaurantId } = req.params;
 
-  const { sort, keyword, star } = req.body;
+  const { sort, keywords, star } = req.body;
   let orderByStr = '';
   let selectStr = orderByStr;
   let sortDirection = '';
   let keywordStr = '';
   let starStr = '';
 
   if (sort === 'highest rating') {
-    orderByStr = 'CAST((reviews.foodrating+reviews.servicerating+reviews.ambiencerating+reviews.valuerating) AS FLOAT)/4';
+    orderByStr = ', CAST((reviews.foodrating+reviews.servicerating+reviews.ambiencerating+reviews.valuerating) AS FLOAT)/4  AS avg_overal_from_reviews';
     sortDirection = ' DESC,';
   }
 
   if (sort === 'lowest rating') {
-    orderByStr = 'CAST((reviews.foodrating+reviews.servicerating+reviews.ambiencerating+reviews.valuerating) AS FLOAT)/4';
+    orderByStr = ', CAST((reviews.foodrating+reviews.servicerating+reviews.ambiencerating+reviews.valuerating) AS FLOAT)/4 AS avg_overal_from_reviews';
     sortDirection = ' ASC,';
   }
 
-  if (keyword !== undefined) {
-    for (let i = 0; i < keyword.length; i ++) {
-      keywordStr += `AND reviews.reviewText LIKE '%${keyword[i]}%' `;
+  if (keywords.length !== 0) {
```

Comment:
if (keywords && keywords.length !== 0)

how do you guarantee keywords won't be undefined?

Reply:
I was thinking to restrict `keywords` to be sent as an empty array by default from the client. But you are right, I should define the condition where `keywords` is undefined.